Source/Core/VideoCommon/TextureDecoder_Common.cpp

```diff
@@ -434,6 +434,11 @@ static inline u32 MakeRGBA(int r, int g, int b, int a)
   return (a << 24) | (b << 16) | (g << 8) | r;
 }
 
+static inline u8 DXTBlend(u8 v1, u8 v2)
+{
+  return ((v1 * 3 + v2 * 5) >> 3);
```

Comment:
this needs an explicit cast to `u8` otherwise it truncates implicitly (arithmetic would promote to int).

```cpp
...
return static_cast<u8>((v1 * 3 + v2 * 5) >> 3);
...
```